src/classes/ADDR_GoogleGeoAPI_Validator.cls

```diff
@@ -161,24 +165,24 @@ public with sharing class ADDR_GoogleGeoAPI_Validator implements ADDR_IValidator
 
                 String verified_street = '';
                 String verified_premise = '';
-                if (googresp.status == 'OK' && googresp.results.size() == 1){
+                if (googresp.status == 'OK' && googresp.results.size() == 1 && !(settings.Reject_Ambiguous_Addresses__c && googresp.results[0].partial_match)){
```

Comment:
I think inline comments only work if you don't have w=1 set.

Line 168 is all ANDs, so the googresp.results.size() has to be 1.